### ISSUE TITLE
Configure IDP sync (organization sync, group sync, role sync) in the dashboard UI

### ISSUE THREAD

**Issue by bpmct** [coder/coder#15290]
## Problem Statement 

IDP Sync UI is runtime configurable and has scheduled improvements.

- Today, managing group/role sync requires a JSON schema (https://github.com/coder/coder/issues/15939)
- Today the UI requires exact knowledge on the claims. (https://github.com/coder/internal/issues/210)

**Comment by jaaydenh** [comment]
@chrifro The IDP sync settings page for groups and roles will need to be redesigned to handle create, update, edit and delete functionality. Its possible the design can mirror that of Organization IdP Sync but I would like to get your feedback to confirm if this should be the case or if there could be an differences in the designs.

**Comment by bpmct** [comment]
Yep, I completely agree. They all 3 should have the same design, but I'm open to identifying opportunities for improvement along the way that we apply to all 3.

**Comment by chrifro** [comment]
>The IDP sync settings page for groups and roles will need to be redesigned to handle create, update, edit and delete functionality.

You're referring to this page, right? Could you share an example screenshot of how it looks right now? I could only find empty states
<img width="1459" alt="Image" src="https://github.com/user-attachments/assets/41b55fef-3d5b-4237-9198-238153614859" />

The IdP Organization Sync settings will look like [this](https://www.figma.com/design/DDdfRMoFIvJo1XkxW1lW11/Idp-Organizations-Sync-Wireframe?node-id=164-2057&t=DscYkToUHOtUkpeZ-1) (currently in implementation)
<img width="748" alt="Image" src="https://github.com/user-attachments/assets/d41c111c-5368-4abb-83ac-10aae59eafe6" />

Would it work to simply add options to edit and update an entry next to the delete button? And to have a create button above the tables?

**Comment by jaaydenh** [comment]
@chrifro you can see a example of idp group sync with data here, https://github.com/coder/coder/pull/14755, right now its only possible to add rows to the table with the cli.

One question is if anything should change regarding the tabs to choose between group and role pages?

I think the area above the table, creating a new row in the table, etc, could be the same as the organization idp sync.

Editing rows is something I have not done for organization id sync settings and I would probably do this separately as well for group/role sync because this requires replacing the table component

**Comment by chrifro** [comment]
>One question is if anything should change regarding the tabs to choose between group and role pages?

Let's replace the blue with a white underline (see [mockup](https://www.figma.com/design/OR75XeUI0Z3ksqt1mHsNQw/Dashboard-v1?node-id=1849-2788&t=BPdcShiz0MKKbG1G-1)). I added an alternative tab component to the design system (see [here](https://www.figma.com/design/WfqIgsTFXN2BscBSSyXWF8/Coder-kit?node-id=1329-6449&t=LaEZvLCAMYxCyam2-1))

>I think the area above the table, creating a new row in the table, etc, could be the same as the organization idp sync.

Agreed

>Editing rows is something I have not done for organization id sync settings and I would probably do this separately as well for group/role sync because this requires replacing the table component

👍 I did not address that yet as you mentioned in a DM that this is not prioritised yet. For now I'm just using a screenshot of the existing table design. I will look at the tables holistically in separate issue.

**Additional changes:**
- I'm using the [code snippet component](https://www.figma.com/design/WfqIgsTFXN2BscBSSyXWF8/Coder-kit?node-id=608-3871&t=LaEZvLCAMYxCyam2-1) for the specific values
- I moved the "showing 2 of 2" below the table and right-aligned it. It now matches other upcoming changes for tables
- not sure where to place "export policy". I assume that it's used rather seldom and placed in underneath the table. Alternatively, we could move it right below the tabs. WDYT?


|new design | old design |
|-|-|
|<img width="715" alt="Image" src="https://github.com/user-attachments/assets/d2d1f4e9-95a2-4549-91a5-6e16a9657976" />| <img width="785" alt="Image" src="https://github.com/user-attachments/assets/cf1033ca-ac8c-47dc-b7f0-d8266f89b9ff" />

**Comment by chrifro** [comment]
>Regarding the code snippet component is this meant just to display data or for editing as well?

Ah I see, I based it on the status quo and at least for me, it's not editable. So this is a new functionality that we want to introduce? 


I'll update the mockup accordingly and turn these fields into form components. Sorry, I didn't get it before that the create, update, edit and delete functionality did refer to this section as well and not just to the table.

**Comment by jaaydenh** [comment]
@chrifro Here is the current state of the UI for groups and roles sync. Let me know your suggestions for the layout. I am still not sure about the best placement for the Export Policy button.

<img width="1178" alt="Image" src="https://github.com/user-attachments/assets/2db86bcf-3670-4082-9506-5a10dc775dd6" />
<img width="1177" alt="Image" src="https://github.com/user-attachments/assets/5d8d5f10-4a05-40cd-9965-792675ace487" />

**Comment by jaaydenh** [comment]
@chrifro from an implementation perspective the save button currently just submits the form with whatever values are in both the sync field and the regex filter. There is no co-dependence between them, they are independent. If there are 2 save buttons this would add some extra complexity because now I would have to make sure the save button for sync field doesn't also save changes for regex filter and vice versa. However, if 2 save buttons feels like a much better UX then it can be done this way.
